charcoal/compare_taxonomy.py

```diff
@@ -160,6 +160,24 @@ def get_genome_taxonomy(matches_filename, genome_sig_filename, provided_lineage,
     if new_siglist:
         siglist = new_siglist
 
+    # remove duplicate signatures in matches
+    md5_list = []
```

Comment:
suggest using a set: `md5_list = set ()`.

Then use `add` instead of `append`, and use ... ok, reading the code below, I think you can just use `seen_md5` without `md5_list`?

Reply:
Ya I think I could just do:
```
        new_siglist = []
        seen_md5 = set()
        for ss in siglist:
            ss_md5 = ss.md5sum()
            if not ss_md5 in seen_md5:
                new_siglist.append(ss)
                seen_md5.add(ss_md5)    
            else:
                print(f'removing a duplicate match: {ss.name}')
        siglist = new_siglist
```

Without all the rest, but my thought was to check first if filtering needed to be done and then filter, instead of filtering all runs through this code. Thoughts?

Reply:
I think shorter code, that's also faster and less memory intensive, is better, unless there's a strong reason otherwise. Not sure that answers the question tho :)

Reply:
okey dokey, went with shorter code in ce5dad1
also added a comment pointing to the sourmash SBT issue so this code chunk could maybe be removed one day :)